src/ucp/core/ucp_am.c

```diff
@@ -626,55 +633,74 @@ static ucs_status_t ucp_am_long_first_handler(void *am_arg, void *am_data,
                                                           f_hdr->super.ep_ptr);
     ucp_ep_ext_proto_t *ep_ext = ucp_ep_ext_proto(ep);
     uint16_t am_id             = f_hdr->super.super.am_id;
+    ucp_recv_desc_t *m_rdesc, *f_rdesc, *data_rdesc;
+    ucs_status_t status, desc_status;
     size_t left;
-    ucp_recv_desc_t *rdesc;
     ucp_ep_h reply_ep;
-    ucp_am_unfinished_t *unfinished;
     ucp_am_mid_hdr_t *m_hdr;
 
-    reply_ep = (f_hdr->super.super.flags & UCP_AM_SEND_REPLY) ?  ep : NULL;
-    left     = f_hdr->total_size - (am_length - sizeof(*f_hdr));
+    left = f_hdr->total_size - (am_length - sizeof(*f_hdr));
 
     if (left == 0) {
         /* Can be a single fragment if send was issued on stub ep */
+        reply_ep = (f_hdr->super.super.flags & UCP_AM_SEND_REPLY) ?  ep : NULL;
```

Comment:
why not pass ep always?

Reply:
so the user would not get it if not requested